crates/extensions/tedge_log_manager/src/actor.rs

```diff
@@ -197,34 +193,37 @@ impl LogManagerActor {
         topic: &str,
         result: UploadResult,
     ) -> Result<(), LogManagementError> {
-        if let Some(mut request) = self.pending_operations.remove(topic) {
-            let topic = Topic::new_unchecked(topic);
-            match result {
-                Ok(response) => {
-                    request.successful();
-
-                    info!("Log request processed for log type: {}.", request.log_type);
-
-                    if let Err(err) = std::fs::remove_file(&response.file_path) {
-                        warn!(
-                            "Failed to remove temporary file {}: {}",
-                            response.file_path, err
-                        )
-                    }
+        let Some(mut request) = self.pending_operations.remove(topic) else {
+            return Ok(());
```

Comment:
Wouldn't it be better to log a warning here before the return, so that if at all this unexpected path is hit in future for some reason, we'd have some clues?